lighthouse-cli/test/smokehouse/lighthouse-runners/devtools.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview A runner that launches Chrome and executes Lighthouse via DevTools.
+ */
+
+import fs from 'fs';
+import os from 'os';
+import {spawn} from 'child_process';
+
+import {LH_ROOT} from '../../../../root.js';
+
+/**
+ * @param {string} command
+ * @param {string[]} args
+ */
+async function spawnAndLog(command, args) {
+  let log = '';
+
+  /** @type {Promise<void>} */
+  const promise = new Promise((resolve, reject) => {
+    const spawnHandle = spawn(command, args);
+    spawnHandle.on('close', code => {
+      if (code === 0) resolve();
+      else reject(new Error(`Command exited with code ${code}`));
+    });
+    spawnHandle.on('error', reject);
+    spawnHandle.stdout.on('data', data => {
+      console.log(data.toString());
```

Comment:
are these lines intended to stay or for debugging?

Reply:
Yes, otherwise the cases where things hang or error aren't printed (the runner interface only allows returning successful logs). and it's nicer to see the output stream in.

Reply:
just checking that it's intentional. The tradeoff is that concurrent runs will be streaming in stuff all interleaved with each other, but that can be helpful, especially when things are new and need to check that it's actually running, etc